frontend/packages/dev-console/src/components/topology/actions/edgeActions.ts

```diff
@@ -0,0 +1,83 @@
+import { KebabOption } from '@console/internal/components/utils/kebab';
+import { modelFor, referenceFor } from '@console/internal/module/k8s';
+import { asAccessReview } from '@console/internal/components/utils';
+import { BaseEdge, Node } from '@console/topology';
+import { getTopologyResourceObject } from '../topology-utils';
+import { removeConnection } from '../components/removeConnection';
+import {
+  TYPE_CONNECTS_TO,
+  TYPE_EVENT_SOURCE,
+  TYPE_EVENT_SOURCE_LINK,
+  TYPE_KNATIVE_REVISION,
+  TYPE_KNATIVE_SERVICE,
+  TYPE_REVISION_TRAFFIC,
+  TYPE_SERVICE_BINDING,
+} from '../const';
+import { moveConnectionModal } from '../components/MoveConnectionModal';
+
+const moveConnection = (edge: BaseEdge, availableTargets: Node[]) => {
+  const resourceObj = getTopologyResourceObject(edge.getSource().getData());
+  const resourceModel = modelFor(referenceFor(resourceObj));
+
+  return {
+    label: 'Move Connector',
+    callback: () => {
+      moveConnectionModal({ edge, availableTargets });
+    },
+    accessReview: asAccessReview(resourceModel, resourceObj, 'delete'),
+  };
+};
+
+const deleteConnection = (edge: BaseEdge) => {
```

Comment:
```suggestion
const deleteConnection = (edge: BaseEdge): KebabOption => {
```

We need to make sure we type output of newly created objects to insure they stay properly typed (if anything in the type changes later).